src/AccessibilityInsights.SharedUx/Controls/ProgressRingControl.xaml.cs

```diff
@@ -1,278 +0,0 @@
-        {
-            var control = d as ProgressRingControl;
-            if (control != null)
-            {
-                control.IsActive = (bool) e.NewValue;
-            }
-        }
-
-        /// <summary>
-        /// WithSound Property Change event handler
-        /// </summary>
-        /// <param name="d"></param>
-        /// <param name="e"></param>
-        private static void WithSoundPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
-        {
-            var control = d as ProgressRingControl;
-            if (control != null)
-            {
-                control.WithSound = (bool)e.NewValue;
-            }
-        }
-
-        /// <summary>
-        /// Private variables used
-        /// </summary>
-        private SoundPlayer player;
-        private Timer timer;
-        private bool isPlayingSound;
-        private bool withSound;
-
-        /// <summary>
-        /// Check if Narrator is running
-        /// </summary>
-        public static bool IsInternalScreenReaderActive()
-        {
-            try
-            {
-                using (RegistryKey key = Registry.CurrentUser.OpenSubKey(Properties.Resources.ProgressRingControl_IsInternalScreenReaderActive_Software_Microsoft_Windows_NT_CurrentVersion_AccessibilityTemp))
-                {
-                    return key?.GetValue("narrator")?.ToString().Equals("1", StringComparison.Ordinal) == true;
-                }
-            }
-#pragma warning disable CA1031 // Do not catch general exception types
-            catch (Exception e)
-            {
-                e.ReportException();
-                // fail silently and we might end up not playing sound
-                return false;
-            }
-#pragma warning restore CA1031 // Do not catch general exception types
-        }
-
-        /// <summary>
-        /// Check if any 3rd party Screen Reader such as Jaws is running
-        /// </summary>
-        public static bool IsExternalScreenReaderActive()
-        {
-            const int SPI_GETSCREENREADER = 0x0046;  // 0x0046 is for SPI_GETSCREENREADER
-            bool active = false;
-            bool success = NativeMethods.SystemParametersInfo(SPI_GETSCREENREADER, 0, out active, 0);
-            return success && active;
-        }
-
-        /// <summary>
-        /// Check if any Screen Reader is running
-        /// </summary>
-        public static bool IsScreenReaderActive()
-        {
-            return (IsInternalScreenReaderActive() || IsExternalScreenReaderActive()) && AutomationPeer.ListenerExists(AutomationEvents.LiveRegionChanged);
-        }
-
-        /// <summary>
-        /// Check if we need to play scanning sound, play sound while meet both conditions below:
-        ///    1. "play sound while scanning" option in settings pannel is on
-        ///    2. At least one AT tool is running
-        /// </summary>
-        public static bool ShouldPlayScannerSound()
```

Comment:
`ShouldPlayScannerSound` in the CommonUxComponents version of `ProgressRingControl` does not access the user preference for `PlayScanningSound`. There is currently no way for CommonUxComponents to access user preferences, and I think it should stay that way. One option is to add a DependencyProperty to `ProgressRingControl` that allows you to pass in the config value from the hosting control.